carma_cooperative_perception/src/msg_conversion.cpp

```diff
@@ -322,19 +322,24 @@ auto to_detection_msg(
     switch (object.object_type) {
       case object.SMALL_VEHICLE:
         detection.motion_model = motion_model_mapping.small_vehicle_model;
+        detection.semantic_class = detection.SEMANTIC_CLASS_SMALL_VEHICLE;
         break;
       case object.LARGE_VEHICLE:
         detection.motion_model = motion_model_mapping.large_vehicle_model;
+        detection.semantic_class = detection.SEMANTIC_CLASS_LARGE_VEHICLE;
         break;
       case object.MOTORCYCLE:
         detection.motion_model = motion_model_mapping.motorcycle_model;
+        detection.semantic_class = detection.SEMANTIC_CLASS_MOTORCYCLE;
         break;
       case object.PEDESTRIAN:
         detection.motion_model = motion_model_mapping.pedestrian_model;
+        detection.semantic_class = detection.SEMANTIC_CLASS_PEDESTRIAN;
         break;
       case object.UNKNOWN:
       default:
         detection.motion_model = motion_model_mapping.unknown_model;
+        detection.semantic_class = detection.SEMANTIC_CLASS_UNKNOWN;
```

Comment:
273-284 lines look similar as well, should we have it there?

Reply:
The semantic classes from SDSMs don't have a great mapping to our sematic classes, but we can do a "best-effort" conversion.

Reply:
The SDSM standard is using J2735 BasicVehicleClass msg (https://github.com/usdot-fhwa-stol/carma-msgs/blob/develop/j2735_msgs/msg/BasicVehicleClass.msg#L103). 
We decided to use the LIGHT_TRUCK_VEHICLE_TYPE_UNKNOWN=20 for the large trucks VRU use case using. So once we support that conversion and message type, we can include that option here. I created an issue:

Reply:
https://github.com/usdot-fhwa-stol/carma-platform/issues/2240